acceptance/new-user/join-button.test.js

```diff
@@ -0,0 +1,64 @@
+/** @jest-environment jsdom */
+const puppeteer = require("puppeteer");
+const URLS = require("../../constants/urls");
+const { JOIN } = URLS;
+const { delay } = require("../../constants/utils");
+
+let browser, page;
+
+async function confirmAlerts() {
+  await page.on("dialog", async (dialog) => {
+    await dialog.accept();
+
+    delay(2000);
+    await page.waitForNavigation();
+    expect(page.url()).toMatch(/login/);
+  });
+}
+
+/* check status*/
+async function checkResponse() {
+  await page.on("response", (response) => {
+    if (response.url().endsWith("/users/self")) {
+      switch (response.status()) {
+        case 401:
+          // user not logged in
+          expect(response.status()).toBe(401);
+          expect(page.url().endsWith("/join")).toBeTruthy();
+          confirmAlerts();
+          break;
+        case 200:
+          // user logged in
+          expect(response.status()).toBe(200);
+          expect(page.url().endsWith("/join")).toBeTruthy();
+          break;
+        default:
+          browser.close();
+      }
+    }
+  });
+}
+
+beforeAll(async () => {
+  browser = await puppeteer.launch({
+    headless: true,
+  });
+
+  const context = await browser.createIncognitoBrowserContext();
+
+  page = await context.newPage();
+});
+
+afterAll(async () => {
+  await browser.close();
+});
```

Comment:
can we move this to top similar to as in the other file

Reply:
I tried but so many errors are coming up. I think it would be best to create a separate issue for this and other tests that need changes and updation.

Reply:
can you create a issue ticket and add the link in which you will be fixing the pending issues.